packages/protocol/scripts/truffle/make-release.ts

```diff
@@ -0,0 +1,219 @@
+/* tslint:disable:no-console */
+import { setAndInitializeImplementation } from '@celo/protocol/lib/proxy-utils'
+import { readdirSync, readJsonSync, writeJsonSync } from 'fs-extra'
+import { ASTDetailedVersionedReport } from '@celo/protocol/lib/compatibility/report'
+import { Address, eqAddress, NULL_ADDRESS } from '@celo/utils/lib/address'
+import { linkedLibraries } from '@celo/protocol/migrationsConfig'
+import { basename, join } from 'path'
+
+/*
+ * A script that reads a backwards compatibility report, deploys changed contracts, and creates
+ * a corresponding JSON file to be proposed with `celocli governance:propose`
+ *
+ * Expects the following flags:
+ *   report: The filepath of the backwards compatibility report
+ *   network: The network for which artifacts should be
+ *
+ * Run using truffle exec, e.g.:
+ * truffle exec scripts/truffle/make-release \
+ *   --network alfajores --build_directory build/alfajores/ --report report.json \
+ *   --initialize_data initialize_data.json --proposal proposal.json
+ *
+ */
+
+class ContractDependencies {
+  dependencies: Map<string, string[]>
+  constructor(linkedLibraries: any) {
+    this.dependencies = new Map()
+    Object.keys(linkedLibraries).forEach((lib: string) => {
+      linkedLibraries[lib].forEach((contract: string) => {
+        if (this.dependencies.has(contract)) {
+          this.dependencies.get(contract).push(lib)
+        } else {
+          this.dependencies.set(contract, [lib])
+        }
+      })
+    })
+  }
+
+  public get = (contract: string): string[] => {
+    return this.dependencies.has(contract) ? this.dependencies.get(contract) : []
+  }
+}
+
+class ContractAddresses {
+  constructor(public addresses: Map<string, Address>) {}
+  static async create(contracts: string[], registry: any) {
+    const addresses = new Map()
+    contracts.forEach(async (contract: string) => {
+      const registeredAddress = await registry.getAddressForString(contract)
+      if (!eqAddress(registeredAddress, NULL_ADDRESS)) {
+        addresses.set(contract, registeredAddress)
+      }
+    })
+    return new ContractAddresses(addresses)
+  }
+
+  public get = (contract: string): Address => {
+    if (this.addresses.has(contract)) {
+      return this.addresses.get(contract)
+    } else {
+      throw new Error(`Unable to find address for ${contract}`)
+    }
+  }
+
+  public set = (contract: string, address: Address) => {
+    this.addresses.set(contract, address)
+  }
+}
+
+module.exports = async (callback: (error?: any) => number) => {
```

Comment:
Might be out of scope, but it would be cool to have a `dryRun` option that outputs the transactions this script would do when not dry running

Reply:
Yeah I thought about that. It's a bit tricky since we depend on the addresses of the contracts we deploy. Let me see what I can do.